### ISSUE TITLE
Error while implementing RZGate with single open control

### ISSUE THREAD

**Issue by TharrmashasthaPV** [Qiskit/qiskit-terra#4138]
### Information

- **Qiskit Terra version**: 0.14.0.dev0+09240bb
- **Python version**: 3.6
- **Operating system**: Ubuntu 18.04

### What is the current behavior?

Receiving CircuitError: "Register size must be positive (int '0' was provided)" error while trying to append RZGate to a quantum circuit with a single open control.

### Steps to reproduce the problem

```
qc = QuantumCircuit(2)
qc.append(RZGate(pi/6).control(ctrl_state='0'),[0,1])
```

### What is the expected behavior?

The implementation should not raise an error.

**Comment by TharrmashasthaPV** [comment]
The Traceback is as below:

```
CircuitError                              Traceback (most recent call last)
<ipython-input-5-6d085c8b3a87> in <module>
      1 qc = QuantumCircuit(2)
----> 2 qc.append(RZGate(pi/6).control(ctrl_state='0'),[0,1])

~/qiskit_issues/qiskit-terra/qiskit/extensions/standard/rz.py in control(self, num_ctrl_qubits, label, ctrl_state)
     93                 return CRZGate(self.params[0])
     94         return super().control(num_ctrl_qubits=num_ctrl_qubits, label=label,
---> 95                                ctrl_state=ctrl_state)
     96 
     97     def inverse(self):

~/qiskit_issues/qiskit-terra/qiskit/circuit/gate.py in control(self, num_ctrl_qubits, label, ctrl_state)
    125         # pylint: disable=cyclic-import
    126         from .add_control import add_control
--> 127         return add_control(self, num_ctrl_qubits, label, ctrl_state)
    128 
    129     @staticmethod

~/qiskit_issues/qiskit-terra/qiskit/circuit/add_control.py in add_control(operation, num_ctrl_qubits, label, ctrl_state)
     56         # gets converted to a circuit before becoming a generic Gate object.
     57         cgate = standard.CRZGate(*operation.params)
---> 58         return cgate.control(num_ctrl_qubits - 1)
     59     if isinstance(operation, UnitaryGate):
     60         # attempt decomposition

~/qiskit_issues/qiskit-terra/qiskit/circuit/gate.py in control(self, num_ctrl_qubits, label, ctrl_state)
    125         # pylint: disable=cyclic-import
    126         from .add_control import add_control
--> 127         return add_control(self, num_ctrl_qubits, label, ctrl_state)
    128 
    129     @staticmethod

~/qiskit_issues/qiskit-terra/qiskit/circuit/add_control.py in add_control(operation, num_ctrl_qubits, label, ctrl_state)
     61         operation._define()
     62     return control(operation, num_ctrl_qubits=num_ctrl_qubits, label=label,
---> 63                    ctrl_state=ctrl_state)
     64 
     65 

~/qiskit_issues/qiskit-terra/qiskit/circuit/add_control.py in control(operation, num_ctrl_qubits, label, ctrl_state)
     91     import qiskit.extensions.standard.multi_control_rotation_gates
     92 
---> 93     q_control = QuantumRegister(num_ctrl_qubits, name='control')
     94     q_target = QuantumRegister(operation.num_qubits, name='target')
     95     q_ancillae = None  # TODO: add

~/qiskit_issues/qiskit-terra/qiskit/circuit/register.py in __init__(self, size, name)
     47         if size <= 0:
     48             raise CircuitError("Register size must be positive (%s '%s' was provided)"
---> 49                                % (type(size).__name__, size))
     50 
     51         # validate (or cast) name

CircuitError: "Register size must be positive (int '0' was provided)"
```

**Comment by ewinston** [comment]
This will be resolved by #4076.
